### ISSUE TITLE
A problem of Ibert IntSoftmax

### ISSUE THREAD

**Issue by baodii** [huggingface/transformers#11755]
## Environment info

- `transformers` version:
- `transformers` version: 4.6.0.dev0
- Platform: Linux-4.15.0-122-generic-x86_64-with-glibc2.10
- Python version: 3.8.1
- PyTorch version (GPU?): 1.9.0a0+git3c87fe9 (False)
- Tensorflow version (GPU?): not installed (NA)
- Using GPU in script?: No
- Using distributed or parallel set-up in script?: yes
- 
### Who can help

@kssteven418 

## Information

Model I am using (Bert, XLNet ...):

The problem arises when using:
* [ ] the official example scripts: (give details below)
* [ ] my own modified scripts: (give details below)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [ ] my own task or dataset: (give details below)

## To reproduce

1.
2.
3.

## Expected behavior

Hi, I have found a strange thing in the IntSoftmax class of Ibert.

def forward(self, x, scaling_factor):
        if not self.quant_mode:
            return nn.Softmax(dim=-1)(x), None

        x_int = x / scaling_factor

        x_int_max, _ = x_int.max(dim=-1, keepdim=True)
        x_int = x_int - x_int_max
        exp_int, exp_scaling_factor = self.int_exp(x_int, scaling_factor)

        # Avoid overflow
        exp, exp_scaling_factor = self.act(exp_int, exp_scaling_factor)
        exp_int = exp / exp_scaling_factor

        exp_int_sum = exp_int.sum(dim=-1, keepdim=True)
        factor = floor_ste.apply(2 ** self.max_bit / exp_int_sum)
        exp_int = floor_ste.apply(exp_int * factor / 2 ** (self.max_bit - self.output_bit))
        scaling_factor = 1 / 2 ** self.output_bit
        return exp_int * scaling_factor, scaling_factor

The code above is the forward func of IntSoftmax. And the problem is that `exp, exp_scaling_factor = self.act(exp_int, exp_scaling_factor)` in which self.act is an instance of QuantAct of which the input should be real number, but the exp_int is a quant int number. Although the result of the trained model word well, I think this is not right.
